core/src/main/java/com/graphhopper/routing/util/spatialrules/SpatialRule.java

```diff
@@ -29,12 +29,17 @@
  */
 public interface SpatialRule {
 
-    double getMaxSpeed(ReaderWay readerWay, String transportationMode);
+    String MAXSPEED_KEY = "max_speed";
+    String ACCESS_KEY = "access";
 
-    AccessValue isAccessible(ReaderWay readerWay, String transportationMode);
+    Object getObject(String key, ReaderWay readerWay, String transportationMode, Object _default);
+
+    <T> T get(String key, ReaderWay readerWay, String transportationMode, T _default);
 
     List<Polygon> getBorders();
+
     SpatialRule setBorders(List<Polygon> borders);
+
     SpatialRule addBorder(Polygon polygon);
 
     String getCountryIsoA3Name();
```

Comment:
Ok, for now I renamed the method to getUniqueName, so we are already more generic at this point. As you said, the rest can be changed once we have a better feeling for the use cases.